src/controller_examples/rabbitmq_controller/resource/default_user_secret.rs

```diff
@@ -0,0 +1,97 @@
+        KubeGetRequest {
+            api_resource: Secret::api_resource(),
+            name: make_default_user_secret_name(rabbitmq),
+            namespace: rabbitmq.namespace().unwrap(),
+        }
+    }
+
+    fn make(rabbitmq: &RabbitmqCluster, state: &RabbitmqReconcileState) -> Result<DynamicObject, RabbitmqError> {
+        Ok(make_default_user_secret(rabbitmq).marshal())
+    }
+
+    fn update(rabbitmq: &RabbitmqCluster, state: &RabbitmqReconcileState, obj: DynamicObject) -> Result<DynamicObject, RabbitmqError> {
+        let secret = Secret::unmarshal(obj);
+        if secret.is_ok() {
+            Ok(update_default_user_secret(rabbitmq, secret.unwrap()).marshal())
+        } else {
+            Err(RabbitmqError::Error)
+        }
+    }
+
+    fn state_after_create_or_update(obj: DynamicObject, state: RabbitmqReconcileState) -> (res: Result<RabbitmqReconcileState, RabbitmqError>) {
+        let secret = Secret::unmarshal(obj);
+        if secret.is_ok() {
+            Ok(RabbitmqReconcileState {
+                reconcile_step: RabbitmqReconcileStep::AfterKRequestStep(ActionKind::Get, ResourceKind::PluginsConfigMap),
+                ..state
+            })
+        } else {
+            Err(RabbitmqError::Error)
+        }
+    }
+
+    open spec fn get_request_view(rabbitmq: RabbitmqClusterView) -> GetRequest {
+        GetRequest { key: resource::make_default_user_secret_key(rabbitmq) }
+    }
+
+    open spec fn make_view(rabbitmq: RabbitmqClusterView, state: RabbitmqReconcileStateView) -> Result<DynamicObjectView, RabbitmqError> {
+        Ok(resource::make_default_user_secret(rabbitmq).marshal())
+    }
+
+    open spec fn update_view(rabbitmq: RabbitmqClusterView, state: RabbitmqReconcileStateView, obj: DynamicObjectView) -> Result<DynamicObjectView, RabbitmqError> {
+        let secret = SecretView::unmarshal(obj);
+        if secret.is_Ok() {
+            Ok(resource::update_default_user_secret(rabbitmq, secret.get_Ok_0()).marshal())
+        } else {
+            Err(RabbitmqError::Error)
+        }
+    }
+
+    open spec fn state_after_create_or_update_view(obj: DynamicObjectView, state: RabbitmqReconcileStateView) -> (res: Result<RabbitmqReconcileStateView, RabbitmqError>) {
+        let sts = SecretView::unmarshal(obj);
+        if sts.is_Ok() {
+            Ok(RabbitmqReconcileStateView {
+                reconcile_step: RabbitmqReconcileStep::AfterKRequestStep(ActionKind::Get, ResourceKind::PluginsConfigMap),
+                ..state
+            })
+        } else {
+            Err(RabbitmqError::Error)
+        }
+    }
+
+    open spec fn resource_state_matches(rabbitmq: RabbitmqClusterView, obj: DynamicObjectView) -> bool {
+        &&& SecretView::unmarshal(obj).is_Ok()
+        &&& SecretView::unmarshal(obj).get_Ok_0().data == resource::make_default_user_secret(rabbitmq).data
+        &&& SecretView::unmarshal(obj).get_Ok_0().type_ == resource::make_default_user_secret(rabbitmq).type_
```

Comment:
I removed `type_` from Secret in the main branch. You can rebase main and remove this type_ here.